src/components/Permissions/AccessRightsSection.jsx

```diff
@@ -64,44 +75,126 @@ const AccessRightsSection = ({
-          const iconName = getPermissionIconName(type)
-          return (
-            <div key={name}>
-              <ListItem
-                button
-                component="a"
-                href={`/#${routes.appList}/${slugName}/${name}`}
-              >
-                <ListItemIcon>
-                  <Icon
-                    icon={
-                      require(`cozy-ui/transpiled/react/Icons/${iconName}`)
-                        .default
-                    }
-                    size={mediumSize}
-                  />
-                </ListItemIcon>
-                <ListItemText
-                  primary={title}
-                  secondary={t(displayPermissions(verbs))}
-                />
-                <ListItemSecondaryAction>
-                  <Icon
-                    icon={RightIcon}
-                    size={smallSize}
-                    className="u-mr-1"
-                    style={{ color: 'var(--secondaryTextColor)' }}
+        {sortedPermissionsByName.map(
+          ({ description, name, title, verbs, type }, index) => {
+            const iconName = getPermissionIconName(type)
+            return (
+              <div key={name}>
+                <ListItem
+                  button
+                  component="a"
+                  onClick={() =>
+                    openModal({
+                      description,
+                      name,
+                      title,
+                      verbs,
+                      type,
+                      isRemoteDoctypes
+                    })
+                  }
+                >
+                  <ListItemIcon>
+                    <Icon
+                      icon={
+                        require(`cozy-ui/transpiled/react/Icons/${iconName}`)
+                          .default
+                      }
+                      size={mediumSize}
+                    />
+                  </ListItemIcon>
+                  <ListItemText
+                    primary={title}
+                    secondary={t(displayPermissions(verbs))}
                   />
-                </ListItemSecondaryAction>
-              </ListItem>
-              {index !== sortedPermissionsByName.length - 1 && (
-                <Divider variant="inset" />
-              )}
-            </div>
-          )
-        })}
+                  <ListItemSecondaryAction>
+                    <Icon
+                      icon={RightIcon}
+                      size={smallSize}
+                      className="u-mr-1"
+                      style={{ color: 'var(--secondaryTextColor)' }}
+                    />
+                  </ListItemSecondaryAction>
+                </ListItem>
+                {index !== sortedPermissionsByName.length - 1 && (
+                  <Divider variant="inset" />
+                )}
+              </div>
+            )
+          }
+        )}
       </NavigationListSection>
+      <DialogComponent
+        open={modalOpened}
```

Comment:
plutôt que d'utiliser le bool d'open on conditionne carrément la Dialog, en forçant le open à true. Sinon le composant se monte même s'il n'est pas visible, ce qui peut poser problème si le composant a des var undefined et qu'il faut attendre quelques render pour avoir les valeurs. C'est une bonne habitude à prendre...

Reply:
Corrigé: https://github.com/cozy/cozy-settings/pull/546/commits/17df1d427e9e4c6a17af78d6611655cbcc81c330